IPs/RSKIP350.md

```diff
@@ -0,0 +1,199 @@
+
+The method does not return any value.
+The gas cost of the `rejectPendingPegouts` method is set to 5000*maxEntries.
+
+## getPendingPegoutState
+
+This method returns the peg-out request entry serialized. If the `requestId` does not exists, the returned data is empty.
+
+## Internal Storage
+
+Entries in the `pendingPegoutRequests` are stored as independent cells in the Bridge contract storage. To derive the storage address, the message `("pendingAcceptance-" + Hex(requestId))` is hashed with Keccak256. The function Hex() returns the 64-byte lowercase hexadecimal representation of the hash without the "0x" prefix.
+
+The peg-out request payload consist of:
+
+* `senderAddress`
+* `amount`
+* `rskTxHash`
+* `acceptanceCount`
+* `rejectionCount`
+* `voted`
+* `previous`
+* `requestEpoch`
+
+The `acceptanceCount` and `rejectionCount` fields are stored as an `int16`.
+The `voted ` field is a bit-vector of fixed size 16 bytes (supports a maximum of 128 pegnatories). The most significant bit of the first byte corresponds to the first pegnatory.
+The `previous` is an `uint256`. All fields are fixed-length.
+
+## Migration
+During powpeg migration, when the new federation is activated, if there are peg-out request outstanding in the map then:
+
+* All entries have their fields `acceptanceCount`, `rejectionCount` and `voted` zeroed. The fields are zeroed even if they had already reached their acceptance or rejection thresholds.
+* The event `pendingPegouts(lastRequestId)` is emmitted.
+
+The new powpeg must approve or reject the prexisting set of peg-out requests. 
+
+## Peg-out Batching
+
+The batch creation time is redefined in terms of epochs, specifically 3 epochs. Every 3 epochs the bridge will attempt to create a new batch. 
+
+## Peg-out Fee Estimation
+
+Peg-out fees will be estimated according to the number of transactions queued in the batch, as it is today, and not conting the transactions that are still in the `pendingPegoutRequests` map.
+
+# Rationale
+
+This proposal is deisgned to enable batch accepts and rejections, because forcing individual votes can be very costly to pegnatories in terms of gas spent. To enalbe batch votes, it is important that the batch voted cannot change between the time the pegnatory is notified of the batch and the time the vote transaction is included in the blockchain. That's why votes apply to all peg-out requests until a specified point in the batch. An alternative would be to divide the time in two phases, one that builds a batch, and another one for voting. I nthe voting phase, new peg-outs are queued for the following batch. However, this design still has the problem that blockchain reorganizations can change the content of batches, and pegnatories votes should never mutate with block reorganizations. Therefore the votes must include either the id of a chain of peg-outs, or a block hash (which indirectly also fixes all previous peg-out requests). 
```

Comment:
```suggestion
This proposal is deisgned to enable batch accepts and rejections, because forcing individual votes can be very costly to pegnatories in terms of gas spent. To enalbe batch votes, it is important that the batch voted cannot change between the time the pegnatory is notified of the batch and the time the vote transaction is included in the blockchain. That's why votes apply to all peg-out requests until a specified point in the batch. An alternative would be to divide the time in two phases, one that builds a batch, and another one for voting. In the voting phase, new peg-outs are queued for the following batch. However, this design still has the problem that blockchain reorganizations can change the content of batches, and pegnatories votes should never mutate with block reorganizations. Therefore the votes must include either the id of a chain of peg-outs, or a block hash (which indirectly also fixes all previous peg-out requests). 
```